<launch>
    
    <!--- Sim Time -->
    <param name="/use_sim_time" value="true" />
    <!--- Imu topic -->
    <remap from="/imu" to="/imu/data"/>
    <!--- Run Rviz-->
    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find lego_loam)/launch/test.rviz" />

    <!--- TF -->
    <node pkg="tf" type="static_transform_publisher" name="camera_init_to_map"  args="0 0 0 1.570795   0        1.570795 /map    /camera_init 10" />
    <node pkg="tf" type="static_transform_publisher" name="base_link_to_camera" args="0 0 0 -1.570795 -1.570795 0        /camera /base_link   10" />

    <!--- LeGO-LOAM -->    
    <node pkg="lego_loam" type="imageProjection"    name="imageProjection"    output="screen">
        <rosparam command="load" file="$(find lego_loam)/params/legoloam.yaml" />
    </node>
    
    <node pkg="lego_loam" type="featureAssociation" name="featureAssociation" output="screen">
        <rosparam command="load" file="$(find lego_loam)/params/legoloam.yaml" />
    </node>
    <node pkg="lego_loam" type="mapOptmization"     name="mapOptmization"     output="screen">
        <rosparam command="load" file="$(find lego_loam)/params/legoloam.yaml" />
    </node>
    
    <node pkg="lego_loam" type="transformFusion"    name="transformFusion"    output="screen">
        <rosparam command="load" file="$(find lego_loam)/params/legoloam.yaml" />
    </node>
</launch>
